src/clapsolver/clap.py

```diff
@@ -22,14 +22,138 @@ def cost(i, j, costs):
         The total cost of the linear assignment problem solution under the
         constraint that row i is assigned to column j.
     """
-    costs = np.array(costs)
-    n_rows, n_cols = costs.shape
+    cost_matrix = np.array(cost_matrix)
+
+    if cost_matrix[i, j] == np.Inf or cost_matrix[i, j] == np.NAN:
+        return cost_matrix[i, j]
+
+    n_rows, n_cols = cost_matrix.shape
 
     # Cost matrix omitting the row and column corresponding to the constraint.
-    sub_costs = costs[~one_hot(i, n_rows), :][:, ~one_hot(j, n_cols)]
+    sub_cost_matrix = cost_matrix[~one_hot(i, n_rows), :][:, ~one_hot(j, n_cols)]
 
     # Lsap solution for the submatrix.
-    sub_row_ind, sub_col_ind = lap(sub_costs)
+    sub_row_ind, sub_col_ind = lap(sub_cost_matrix)
 
     # Total cost is that of the submatrix lsap plus the cost of the constraint.
-    return sub_costs[sub_row_ind, sub_col_ind].sum() + costs[i, j]
+    return sub_cost_matrix[sub_row_ind, sub_col_ind].sum() + cost_matrix[i, j]
+
+
+def costs(cost_matrix):
+    """Solve a constrained linear sum assignment problem for each entry.
+
+    The output of this function is equivalent to, but significantly more
+    efficient than,
+
+    >>> def costs(cost_matrix):
+    ...     total_costs = np.empty_like(cost_matrix)
+    ...     for i, j in np.ndindex(*cost_matrix.shape):
+    ...         total_costs[i, j] = cost(i, j, cost_matrix)
```

Comment:
Should this be `cost(i, j, cost_matrix)` or `clap.cost(i, j, cost_matrix)`? I lean toward the latter since the example usage in `README.md` uses
```python
>>> from clapsolver import clap
>>> clap.cost(i, j, cost_matrix)
```